### PR TITLE
[Merged by Bors] - Increase log volume size limit

### PR DESCRIPTION
**Pull request by razvan**
# Description

*Please add a description here. This will become the commit message of the merge request later.*

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [x] Changes are OpenShift compatible
- [x] CRD changes approved
- [x] Helm chart can be installed and deployed operator works
- [x] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [x] Code contains useful comments
- [x] (Integration-)Test cases added
- [x] Documentation added or updated
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,11 +15,12 @@ All notable changes to this project will be documented in this file.
 
 ### Changed
 
-- `operator-rs` `0.38.0` -> `0.41.0` ([#235]).
+- `operator-rs` `0.38.0` -> `0.44.0` ([#235]), ([#259]).
```
Comment (siegfriedweber) [inline CHANGELOG.md:18]:
```suggestion
- `operator-rs` `0.38.0` -> `0.44.0` ([#235], [#259]).
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -15,11 +15,12 @@ All notable changes to this project will be documented in this file.
...
 - Use 0.0.0-dev product images for testing ([#236])
 - Use testing-tools 0.2.0 ([#236])
 - Run as root group ([#241]).
 - Added kuttl test suites ([#252])
+- Increase the size limit of log volumes (#[259])
```
Comment (siegfriedweber) [inline CHANGELOG.md:23]: Could be moved to the Fixed section.

File: rust/crd/src/constants.rs [hunk 3]
```diff
@@ -40,10 +42,10 @@ pub const MIN_MEMORY_OVERHEAD: u32 = 384;
 pub const JVM_OVERHEAD_FACTOR: f32 = 0.1;
 pub const NON_JVM_OVERHEAD_FACTOR: f32 = 0.4;
 
-pub const MAX_SPARK_LOG_FILES_SIZE_IN_MIB: u32 = 10;
-pub const MAX_INIT_CONTAINER_LOG_FILES_SIZE_IN_MIB: u32 = 1;
```
Comment (siegfriedweber) [inline rust/crd/src/constants.rs:44]: The log file size for the init container should still be listed.

**Review by siegfriedweber — CHANGES_REQUESTED** [review]

File: rust/crd/src/constants.rs (hunk 3 above)
Comment (razvan) [reply rust/crd/src/constants.rs:44]: I think this is not needed.

`SparkApplications` have up to three init containers but none of them use log4j for logging:
* `job` - copies the user jar-file to a dedicated place
* `requirements` - installs Python pip packages
* `tls` - generates a trust store

The spark history server has no init (or prepare) containers.

**Comment by razvan** [comment]
https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/spark-k8s-operator-it-custom/91/

File: rust/crd/src/constants.rs (hunk 3 above)
Comment (siegfriedweber) [reply rust/crd/src/constants.rs:44]: `capture_shell_output` is used in the init containers, so log files are written even if they contain only few kilobytes.

Comment (razvan) [reply rust/crd/src/constants.rs:44]: hm ok, wan't clear about that. Then I'll set it to 3Mib for consistency

Comment (razvan) [reply rust/crd/src/constants.rs:44]: Done https://github.com/stackabletech/spark-k8s-operator/pull/259/commits/19c92641f8b6e61e7de6dcca602d15492ad6d317

**Review by siegfriedweber — APPROVED** [review]

**Comment by razvan** [comment]
bors merge

**Pull request closed without merging**